packages/components/color-picker/tests/saturation-slider.test.tsx

```diff
@@ -1,7 +1,36 @@
-import { fireEvent, render, screen } from "@yamada-ui/test"
+import { drag, fireEvent, render, screen, waitFor } from "@yamada-ui/test"
+import type { Hsv } from "../src"
 import { SaturationSlider } from "../src"
 
 describe("<SaturationSlider />", () => {
+  beforeAll(() => {
+    Object.defineProperties(MouseEvent.prototype, {
+      pageX: {
+        get() {
+          return this.clientX
+        },
+        configurable: true,
+      },
+      pageY: {
+        get() {
+          return this.clientY
+        },
+        configurable: true,
+      },
+    })
+  })
+
+  afterAll(() => {
+    Object.defineProperty(MouseEvent.prototype, "pageX", {
+      value: undefined,
+      configurable: true,
+    })
+    Object.defineProperty(MouseEvent.prototype, "pageY", {
+      value: undefined,
+      configurable: true,
+    })
+  })
+
```

Comment:
In my personal opinion, it would be better to modify the code to use beforeEach and afterEach instead of beforeAll and afterAll. This will help ensure that each test runs independently and maintains its own state, making it easier to debug and maintain in the long run.


```suggestion
  let originalPageX: PropertyDescriptor | undefined;
  let originalPageY: PropertyDescriptor | undefined;

  beforeEach(() => {
    originalPageX = Object.getOwnPropertyDescriptor(MouseEvent.prototype, 'pageX');
    originalPageY = Object.getOwnPropertyDescriptor(MouseEvent.prototype, 'pageY');

    Object.defineProperty(MouseEvent.prototype, 'pageX', {
      get() {
        return this.clientX;
      },
      configurable: true,
    });

    Object.defineProperty(MouseEvent.prototype, 'pageY', {
      get() {
        return this.clientY;
      },
      configurable: true,
    });
  });

  afterEach(() => {
    if (originalPageX) {
      Object.defineProperty(MouseEvent.prototype, 'pageX', originalPageX);
    }
    if (originalPageY) {
      Object.defineProperty(MouseEvent.prototype, 'pageY', originalPageY);
    }
  });

```